### PR TITLE
Issue #13321: Kill mutation for OrderedPropertiesCheck-2

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13321: Kill mutation for OrderedPropertiesCheck-2

---------

# Check :- 
https://checkstyle.org/checks/misc/orderedproperties.html#OrderedProperties

-----------

# Mutation :- 
https://github.com/checkstyle/checkstyle/blob/20a52f70548476a4ca3bb0f56e4d8fcd9cd775b7/config/pitest-suppressions/pitest-misc-suppressions.xml#L39-L55

----------

# Explaination 
Test added

---------

Diff Regression config: https://gist.githubusercontent.com/Kevin222004/c3213434af8523d918594b88ce5554f7/raw/3565cfcd975e228af515bb34ce22c83afce0a7c7/opc.xml
Diff Regression projects: https://gist.githubusercontent.com/Kevin222004/9600f179b602d4c971bdb0a050099005/raw/360a95ed7bb60d7a0956e531199d484c4d6f6617/test-projects.properties
Report label: Regression-2

### CODE REVIEW

**Comment by Kevin222004** [comment]
GIthub, generate report

**Comment by Kevin222004** [comment]
Github, generate report

**Comment by Vyom-Yadav** [comment]
@Kevin222004 No rationale?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ private static int getIndex(int startLineNo, FileText fileText, String keyName)
      */
     private static Pattern getKeyPattern(String keyName) {
         final String keyPatternString = "^" + SPACE_PATTERN.matcher(keyName)
-                .replaceAll(Matcher.quoteReplacement("\\\\ ")) + "[\\s:=].*";
+                .replaceAll("") + "[\\s:=].*";
```
Comment (Vyom-Yadav) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java:209]: @Kevin222004 You can have properties with space in them, eg:
```properties
foo\ bar:hoo
```
The whitespace is escaped. You can see https://stackoverflow.com/q/2108103/15412365

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java (hunk 1 above)
Comment (Kevin222004) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/OrderedPropertiesCheck.java]: Thanks @Vyom-Yadav for your help.

**Comment by Kevin222004** [comment]
Test added

**Comment by Kevin222004** [comment]
@romani please re-run failing ci

**Comment by romani** [comment]
Circleci was relaunched

**Review by romani — APPROVED** [review]
Ok to merge

Covered by new test

**Pull request merged**